### PR TITLE
Feature ver grupos en perfil 160652470

### PR DESCRIPTION
**Pull request by luchong12**
This branch adds the endpoint needed to get all the groups that a user belongs to.

### CODE REVIEW

**Comment by luchong12** [comment]
I'm a little bit confused about the comments. The route you both propose ('/users/:user_id/groups') is the one I am already creating. I just followed the "Nested Resources" section of the rails guide: https://guides.rubyonrails.org/routing.html#nested-resources

where it says that this:
```
resources :users, only: [:create, :show, :update] do
    resources :groups, only: [:index]
end
```
creates an endpoint in '/users/:user_id/groups' handled by the groups controller, which to me makes sense because even though you are using the user_id to filter them, the important resource here are the groups.

If this still doesn't convince you just tell me and I'll change it right away, and don't worry about being honest, its better that way 👍

PS: Specs arent done yet, working on it

**Comment by Gimli76** [comment]
> I'm a little bit confused about the comments. The route you both propose ('/users/:user_id/groups') is the one I am already creating. I just followed the "Nested Resources" section of the rails guide: https://guides.rubyonrails.org/routing.html#nested-resources
> 
> where it says that this:
> 
> ```
> resources :users, only: [:create, :show, :update] do
>     resources :groups, only: [:index]
> end
> ```
> 
> creates an endpoint in '/users/:user_id/groups' handled by the groups controller, which to me makes sense because even though you are using the user_id to filter them, the important resource here are the groups.
> 
> If this still doesn't convince you just tell me and I'll change it right away, and don't worry about being honest, its better that way +1
> 
> PS: Specs arent done yet, working on it

That's right! My bad, for some reason I misunderstood the routes. Also taking another look at the docs seem that "the right way" is to do it in the groups controller.

What still feels odd is that the logic is in the index method, can't we separate that logic from the index method?

Sorry for the bother.

**Comment by luchong12** [comment]
After I wrote the comment I kept thinking about the same thing, I'm going to see if I can change the method that handles the request somehow. Thanks for the quick answer!

File: spec/controllers/groups_controller_spec.rb [hunk 1]
```diff
@@ -49,4 +49,14 @@
...
     end
   end
+
+  describe "GET #index" do
+    context "with valid user_id param" do
+      it "returns a success response" do
+        allow(controller).to receive(:authorize)
+        get :index, params: { user_id: user.to_param }
+        expect(response).to be_successful
+      end
```
Comment (joaquinvillar) [inline spec/controllers/groups_controller_spec.rb]: This is only testing if `params[:user_id].blank?` is `false` and never if it is `true`, and there is no specs for the resulting data in both cases you can replicate the index specs of user controller.

**Pull request merged**